src/model/NPMResolver.ts

```diff
@@ -329,22 +329,30 @@ export class NPMResolver {
     private initRegistry(): void {
         console.info("⟲ Initialising cplace jfrog registry for NPM");
         const registry = '//cplace.jfrog.io/artifactory/api/npm/cplace-npm-local/';
-        const gradleProps: string = (execSync(path.join(this.mainRepo, 'gradlew properties')) || '').toString();
+        
+        const npmConfig: string = execSync('npm config ls -l').toString();
+        const npmrcPath: string | undefined = (npmConfig.match(/userconfig *= *".*"/gi) || [])[0];
+        const cleanNpmrcPath: string = npmrcPath.replace(/userconfig *= */, '').replace(/"/gi, '').replace(/\\\\/g, '\\');
 
-        if (!gradleProps) {
-            console.error(cred`✗`, 'Gradle Properties not found');
+        if (!cleanNpmrcPath) {
+            console.error(cred`✗`, 'No user npmrc found');
+            return;
+        }
+
+        if(process.env.ENV_CPLACE_ARTIFACTORY_ACTOR && process.env.ENV_CPLACE_ARTIFACTORY_TOKEN) {
+            this.writeNPMRC(registry, cleanNpmrcPath, Buffer.from(`${process.env.ENV_CPLACE_ARTIFACTORY_ACTOR}:${process.env.ENV_CPLACE_ARTIFACTORY_TOKEN}`).toString('base64'), process.env.ENV_CPLACE_ARTIFACTORY_ACTOR);
```

Comment:
For non docker ci-s (like a Jenkins could be), the `const isConfigured: boolean = currentNpmrcConfig.includes(registry);` check could also make sense here.